### PR TITLE
feat: Document all differences into rpc methods WRT standard

### PR DESCRIPTION
**Pull request by rodolfopietro97**
# Description

The most hated work of a software developer. The documentation 😅😂

P.S.
Please guys can you take a look? My english is not the best

@victhorbi @fabiorigam @Valazan @freemanzMrojo @lucanicoladebiasi

### CODE REVIEW

**Review by fabiorigam — CHANGES_REQUESTED** [review]

File: packages/network/src/provider/utils/rpc-mapper/methods/eth_getBlockByHash/eth_getBlockByHash.ts [hunk 1]
```diff
@@ -12,12 +12,12 @@ import { type ThorClient } from '../../../../../thor-client';
...
  * RPC Method eth_getBlockByHash implementation
  *
- * @link [eth_getBlockByHash](https://docs.infura.io/networks/ethereum/json-rpc-methods/eth_getblockbyhash)
+ * @link [eth_getBlockByHash](https://ethereum.github.io/execution-apis/api-documentation/)
  *
  * @param thorClient - The thor client instance to use.
  * @param params - The standard array of rpc call parameters.
  *                 * params[0]: The block hash of block to get.
- *                 * params[1]: The transaction detail flag. If true, the block will contain the transaction details, otherwise it will only contain the transaction hashes.
+ *                 * params[1]: The transaction hydrated detail flag. If true, the block will contain the transaction details, otherwise it will only contain the transaction hashes.
```
Comment (leszek-vechain) [inline packages/network/src/provider/utils/rpc-mapper/methods/eth_getBlockByHash/eth_getBlockByHash.ts:20]: not sure if hydrated is ok in that context

File: docs/provider.md [hunk 2]
```diff
@@ -81,3 +88,511 @@ const rpcCallChainId = await provider.request({
...
+- Differences with RPC Standard: No
+
+### eth_getBlockTransactionCountByNumber
+
+- Implemented: Yes
+- Differences with RPC Standard: Some differences related to the block number tag.
+
+#### Notes
+
+Vechain supports only the block number tags 'latest' and 'finalized'. 'earliest', 'safe', and 'pending' are not
```
Comment (leszek-vechain) [inline docs/provider.md:316]: we also support these: best stands for latest block, justified stands for the justified block, finalized stands for the finalized block

File: docs/provider.md [hunk 3]
```diff
@@ -81,3 +88,511 @@ const rpcCallChainId = await provider.request({
...
+- Differences with RPC Standard: Different meaning into vechain blockchain.
+
+#### Notes
+
+Vechain blockchain have different meaning for the nonce. In vechain, the nonce is a random number. Not a progressive
+number.
+Since this RPC Method is mainly used to caluclate the nonce, it will return a random number.
+This is not only for this motivation, but because currently there is no a direct endpoint to calculate the transaction
+count.
+More, this behaviot can be customized by the user.
```
Comment (leszek-vechain) [inline docs/provider.md:399]: typo

File: docs/provider.md [hunk 4]
```diff
@@ -81,3 +88,511 @@ const rpcCallChainId = await provider.request({
...
+
+### eth_estimateGas
+
+- Implemented: Yes
+
+### eth_feeHistory
+
+- Implemented: No
+
+### eth_gasPrice
```
Comment (leszek-vechain) [inline docs/provider.md:259]: did use eth_gasPrice, but I only got this: { jsonrpc: '2.0', result: '0x38d7ea4c68000', id: 1 }

File: docs/provider.md [hunk 5]
```diff
@@ -81,3 +88,511 @@ const rpcCallChainId = await provider.request({
...
+'removed' field is always false.
+
+### eth_getProof
+
+- Implemented: No
+
+### eth_getStorageAt
+
+- Implemented: Yes
+- Differences with RPC Standard: Some small differences related to the block number tag.ùù
```
Comment (leszek-vechain) [inline docs/provider.md:355]: some weird chars at the end of the line

File: docs/provider.md [hunk 6]
```diff
@@ -81,3 +88,511 @@ const rpcCallChainId = await provider.request({
...
+### eth_syncing
+
+- Implemented: Yes
+- Differences with RPC Standard: Some differences related to the output of the function.
+
+#### Notes
+
+The 'startingBlock' field is always '0x0' because it is not supported yet.
+
+The fields 'difficulty','totalDifficulty','uncles','sha3Uncles','nonce','logsBloom','extraData','baseFeePerGas','
```
Comment (leszek-vechain) [inline docs/provider.md:529]: eth_syncing on solo returns this: { jsonrpc: '2.0', result: false, id: 67 }

**Review by fabiorigam — APPROVED** [review]

**Pull request merged**